src/crimson/osd/backfill_state.cc

```diff
@@ -472,7 +500,12 @@ BackfillState::ReplicasScanning::react(ReplicaScanned evt)
     if (waiting_on_backfill.empty()) {
       ceph_assert(backfill_state().peer_backfill_info.size() == \
                   peering_state().get_backfill_targets().size());
-      return transit<Enqueuing>();
+      if (!backfill_state().next) {
+	return transit<Enqueuing>();
+      } else {
+	DEBUGDPP("backfill cancelled, not going Enqueuing", pg());
+	backfill_state().go_enqueuing_on_resume();
+      }
```

Comment:
We can skip resetting cancel_backfill since we will discard this event, right?
```
      if (backfill_state().is_cancelled()) {
	return transit<Enqueuing>();
      } else {
	DEBUGDPP("backfill cancelled, not going Enqueuing", pg());
      }
```

Reply:
Um, not really, we need to know whether we should go to `Enqueuing` directly when the backfill resumes